apiserver/apiserver.go

```diff
@@ -172,6 +184,40 @@ func (c *ServerConfig) pingClock() clock.Clock {
 	return c.PingClock
 }
 
+// RateLimitConfig holds holds parameters to control
+// aspects of rate limiting connections and logins.
+type RateLimitConfig struct {
+	LoginRateLimit  int
+	LoginMinPause   time.Duration
+	LoginMaxPause   time.Duration
+	LoginRetryPause time.Duration
+	ConnMinPause    time.Duration
+	ConnMaxPause    time.Duration
+}
+
+// Validate validates the rate limit configuration.
+func (c RateLimitConfig) Validate() error {
```

Comment:
This configurability isn't for the end user. I kept it as hidden as possible. But I could declare a MaxConnMinPause const.